rfc/wip-tform/index.md

```diff
@@ -0,0 +1,679 @@
+an axis with type `coordinate` or `displacement` respectively for transformations of type `coordinates` or `displacements`. This
+axis MUST be of length equal to the number of axes of the output coordinate system and SHOULD be the last axis (contiguous on
+disk when c-order). The coordinate system MUST also contain an axis identical to every axis of its input coordinate system in
+the same order. As a consequence, the array will have one dimension more than its input coordinate system.
+
+The `i`th value of the array along the `coordinate` or `displacement` axis refers to the displacement
+or coordinate of `i`th output axis. See the example below.
+
+
+`coordinates` and `displacements` transformations are not invertible in general, but implementations MAY approximate their
+inverses. Metadata for these coordinate transforms have the following field: 
+
+<dl>
+  <dt><strong>path</strong></dt>
+  <dd>  The location of the coordinate array in this (or another) container.</dd>
+  <dt><strong>interpolation</strong></dt>
+  <dd>  The `interpolation` attributes MAY be provided. It's value indicates
+        the interpolation to use if transforming points not on the array's discrete grid.
+        Values could be:
+        <ul>
+            <li><code>linear</code> (default)</li>
+            <li><code>nearest</code></li>
+            <li><code>cubic</code></li>
+        </ul></dd>
+</dl>
+
+For both `coordinates` and `displacements`, the array data at referred to by `path` MUST define space and coordinate transform metadata:
+
+* Every axis name in the `coordinateTransform`'s `input` MUST appear in the coordinate system
+* The array dimension corresponding to the `coordinate` or `displacement` axis MUST have length equal to the dimensionality of the `coordinateTransform` `output`
+* SHOULD have a `name` identical to the `name` of the corresponding `coordinateTransform`.
+
+For `coordinates`:
+
+* coordinateSystem metadata MUST have exactly one axis with `"type" : "coordinate"`
+* If a `coordinates`' input space has dimensionality `N`, then the array data at `path` MUST have dimensionality equal to `(N + 1)`.
+* the shape of the array along the "coordinate" axis must be exactly `N`
+
+For `displacements`:
+
+* coordinateSystem metadata MUST have exactly one axis with `"type" : "displacement"`
+* If a `displacements`' input space has dimensionality `N`, then the array data at `path` MUST have dimensionality equal to `(N + 1)`.
+* the shape of the array along the "displacement" axis must be exactly `N`
+* `input` and `output` MUST have an equal number of dimensions.
+
+
+
+#### byDimension {#trafo-byDimension}
+
+`byDimension` transformations build a high dimensional transformation using lower dimensional transformations
+on subsets of dimensions.
+
+<dl>
+  <dt><strong>transformations</strong></dt>
+  <dd>  A list of transformations, each of which applies to a (non-strict) subset of input and output dimensions (axes). 
```

Comment:
I think that maybe these conditions leave some edge cases open:
- the byDimension transformation could have a sub transformation whose output is not in the output axes of the byDimension output. Probably we don't want this.
- there could be multiple subtransformations that share some axes in their input, maybe it's ok, but not sure if we want this.
- there could be some input axes of the parent object that have no corresponding input axis in any of the sub transformations. I think that this is acceptable, but I wanted to highlight it here.

Reply:
> the byDimension transformation could have a sub transformation whose output is not in the output axes of the byDimension
> output. Probably we don't want this.

While I wouldn't recommend that, I don't think this is a problem - it seems unambiguous how to compute the transformation. e.g. 

```
{
  "coordinateSystems": [
    {
      "name": "in",
      "axes": [
        { "name": "j", "type": "array" },
        { "name": "i", "type": "array" }
      ]
    },
    {
      "name": "out",
      "axes": [
        { "name": "y", "type": "array" },
        { "name": "x", "type": "array" }
      ]
    }
  ]
  "coordinateTransformations": [
    {   
      "type": "byDimension",
      "input": "in",
      "output": "out",
      "transformations": [
        {
          "type": "translation",
          "translation": [ 1, 3 ],
          "input": [ "i", "j" ],
          "output": [ "x", "IGNORED" ]
        },
        {
          "type": "scale",
          "scale": [ 2, 3 ],
          "input": [ "i", "j" ],
          "output": [ "IGNORED". "y" ]
        }
      ]

    }
  ]
}
```

> there could be multiple subtransformations that share some axes in their input, maybe it's ok, but not sure if we want this.

Agreed that I would not recommend it, but don't think it is a problem. The above example applies to this as well.


> there could be some input axes of the parent object that have no corresponding input axis in any of the sub transformations. I think that this is acceptable, but I wanted to highlight it here.

Agreed.